src/Resources/config/listener.yml

```diff
@@ -87,3 +87,17 @@ services:
             - [setContainer, ["@service_container"]]
         tags:
             - { name: kernel.event_listener, event: kernel.request, method: onKernelRequest, priority: 24 }
+
+    contao.listener.user_session:
+        class: Contao\CoreBundle\EventListener\UserSessionListener
+        calls:
+            - [setContainer, ["@service_container"]]
+        arguments:
+            - '@session'
+            - '@doctrine.dbal.default_connection'
+        tags:
+            # Priority must be lower than contao.listener.session (64)
+            - { name: kernel.event_listener, event: kernel.request, method: onKernelRequest }
+            # Priority must be higher than the framework bundle save session listener (-1000)
+            - { name: kernel.event_listener, event: kernel.response, method: onKernelResponse }
```

Comment:
I did not copy, no. Actually, this is the whole thing about the problem. We have to read and store away the session into the database before the framework bundle listener closes the session.